src/xpk/core/blueprint/blueprint_generator.py

```diff
@@ -133,48 +144,59 @@ def generate_a3_mega_blueprint(
         },
     )
 
+    reservation_affinity = (
+        {
+            "consume_reservation_type": "NO_RESERVATION",
+            "specific_reservations": [],
+        }
+        if reservation is None
+        else {
+            "consume_reservation_type": "SPECIFIC_RESERVATION",
+            "specific_reservations": [{"name": reservation}],
+        }
+    )
+
     a3_megagpu_pool_0 = DeploymentModule(
         id="a3_megagpu_pool_0",
         source="modules/compute/gke-node-pool",
         use=["gke_cluster", gpu_subnets_name, "group_placement_0"],
         settings={
             "name": f"{cluster_name}-a3-megagpu-pool-0",
-            "machine_type": a3_machine_type,
+            "machine_type": system.gce_machine_type,
             "autoscaling_total_min_nodes": autoscaling_total_min_nodes,
             "initial_node_count": num_nodes,
             "zones": [zone],
             "host_maintenance_interval": "PERIODIC",
+            "reservation_affinity": reservation_affinity,
+            "run_workload_script": False,
         },
         outputs=["instructions"],
     )
-    num_chips = get_num_chips(num_nodes, a3_machine_type)
+    num_chips = num_nodes * system.chips_per_vm
     workload = DeploymentModule(
         id="workload_component_install",
         source="modules/management/kubectl-apply",
         use=["gke_cluster"],
         settings={
             "kueue": {
                 "install": True,
-                "config_path": f'$(ghpc_stage("{cluster_name}-a3-mega-xpk"))/kueue-xpk-configuration.yaml.tftpl',
+                "version": "v0.9.1",  # TAS feature-gates is enabled in CT
```

Comment:
please save version is some global variable

Reply:
I would suggest to treat this as a template constant. We cannot simply allow kueue version change in other place as the config needs to become compatible, too. Generally we can argue to move all "kueue" block including the "Install", "config_path", "version" to a global variables. The same goes with Jobset. I don't think currently we want to make it dynamic now.